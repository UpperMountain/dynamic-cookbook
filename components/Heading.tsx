import React from "react";
import { Text, TextProps } from "react-native";

const sizes = {
  1: 22,
  2: 17
};

export interface Props {
  children: string | React.ReactNode;

  // style overrides
  style?: TextProps["style"];
}

export function Heading1({ children, style }: Props) {
  return <Text style={[{ fontSize: sizes[1] }, style]}>{children}</Text>;
}

export function Heading2({ children, style }: Props) {
  return <Text style={[{ fontSize: sizes[2] }, style]}>{children}</Text>;
}

export default Heading1;
