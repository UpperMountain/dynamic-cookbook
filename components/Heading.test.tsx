import React from "react";
import Heading, { Heading1, Heading2 } from "./Heading";
import renderer from "react-test-renderer";

it("has a Heading which is really Heading1", () => {
  expect(Heading).toBe(Heading1);
});

for (let Cmp of [Heading1, Heading2]) {
  describe(`<${Cmp.name}/>`, () => {
    it("should render without exploding", () => {
      renderer.create(<Cmp>testing</Cmp>).unmount();
    });

    it("should pass snapshot tests", () => {
      const cmp = renderer.create(<Cmp>testing</Cmp>);
      expect(cmp.toJSON()).toMatchSnapshot();
      cmp.unmount();
    });
  });
}
