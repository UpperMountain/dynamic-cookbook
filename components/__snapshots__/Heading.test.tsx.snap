// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading1/> should pass snapshot tests 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 22,
      },
      undefined,
    ]
  }
>
  testing
</Text>
`;

exports[`<Heading2/> should pass snapshot tests 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 17,
      },
      undefined,
    ]
  }
>
  testing
</Text>
`;
